File: 2022/6/solution1.py
filename = "6/input.txt"

def solution(part, size):
    count = 0

    while True:
        errors = 0
        # print(line[count:count+4])
        currpos = line[count:count+size]
        for char in currpos:
            # print(char)
            # print(currpos.count(char))
            if currpos.count(char) > 1:
                errors += 1
                count += 1
                break
        
        if errors == 0:
            break
    
    print("=============================")
    print("Part: " + part)
    print(count+size)
    print("=============================")

with open(filename, 'r') as f:
    line = f.readline()

    solution(part="1", size=4)
    solution(part="2", size=14)

        

File: 2020/lib/2/solution1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func solution1() {
	total := 0

	readFile, err := os.Open("lib/2/input.txt")

	if err != nil {
		fmt.Println(err)
	}
	fileScanner := bufio.NewScanner(readFile)
	fileScanner.Split(bufio.ScanLines)

	for fileScanner.Scan() {
		line := fileScanner.Text()
		splitLine := strings.Split(line, " ")
		allowedUsages := strings.Split(splitLine[0], "-")
		min, _ := strconv.Atoi(allowedUsages[0])
		max, _ := strconv.Atoi(allowedUsages[1])

		usages := strings.Count(splitLine[2], string(splitLine[1][0]))

		if min <= usages && max >= usages {
			total += 1
		}
	}

	totalString := strconv.Itoa(total)

	println("Solution 1 Answer: " + totalString)
}
